internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -0,0 +1,189 @@
+		//the resource is being deleted
+		if utils.ContainsString(resourceRequest.ObjectMeta.Finalizers, finalizerName) {
+			if err := r.deleteOffer(); err != nil {
+				klog.Errorf("Error deleting offer")
+				return ctrl.Result{}, err
+			}
+
+			// remove our finalizer from the list and update it.
+			resourceRequest.ObjectMeta.Finalizers = utils.RemoveString(resourceRequest.ObjectMeta.Finalizers, finalizerName)
+			if err := r.Update(ctx, &resourceRequest); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+	offerList := &sharingv1alpha1.ResourceOfferList{}
+	if err = r.getAllOffers(offerList); err != nil || len(offerList.Items) > 0 {
+		if err != nil {
+			klog.Errorf("Unable to get resource offer list")
+			return ctrl.Result{}, err
+		} else {
+			klog.Error("Resource Already exists")
+			return ctrl.Result{}, nil
+		}
+
+	}
+
+	klog.Infof("Generating Offer...")
+	offer, offerErr := r.generateResourceOffer()
+	if offerErr != nil {
+		if offer != nil {
+			klog.Errorf("unable to create resourceOffer %s", offer.ObjectMeta.Name)
+		}
+		klog.Errorf("Error: %s", offerErr)
+		return ctrl.Result{}, offerErr
+	}
+	klog.Infof("ResourceOffer successfully created")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&discoveryv1alpha1.ResourceRequest{}).
+		Complete(r)
+}
+
+func (r *ResourceRequestReconciler) generateResourceOffer() (*sharingv1alpha1.ResourceOffer, error) {
+	offer := &sharingv1alpha1.ResourceOffer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "liqo",
+			Name:      offerPrefix + r.ClusterId,
+		},
+		Spec: sharingv1alpha1.ResourceOfferSpec{
+			ClusterId: r.ClusterId,
+			Images:    nil,
+			LimitRange: corev1.LimitRangeSpec{
+				Limits: []corev1.LimitRangeItem{
+					{
+						Type:                 "",
+						Max:                  nil,
+						Min:                  nil,
+						Default:              nil,
+						DefaultRequest:       nil,
+						MaxLimitRequestRatio: nil,
+					},
+				},
+			},
+			ResourceQuota: corev1.ResourceQuotaSpec{
+				Hard:          nil,
+				Scopes:        nil,
+				ScopeSelector: nil,
+			},
+			Labels:     nil,
+			Neighbors:  nil,
+			Properties: nil,
+			Prices:     nil,
+			KubeConfigRef: corev1.SecretReference{
+				Namespace: "liqo",
+				Name:      "name",
+			},
+			Timestamp:  metav1.NewTime(time.Now()),
+			TimeToLive: metav1.NewTime(time.Now().Add(30 * time.Minute)),
+		},
+	}
+	// call function to fill resource
+
+	//create empty resource for testing
+	err := r.Create(context.Background(), offer)
+	if err != nil {
+		return offer, err
+	}
+	klog.Infof("Generated Offer: %s", offer.ObjectMeta.Name)
+	return offer, nil
+}
+
+func (r *ResourceRequestReconciler) deleteOffer() error {
+	klog.Info("Deleting offer...")
+	//var offer sharingv1alpha1.ResourceOffer
+	offerList := &sharingv1alpha1.ResourceOfferList{}
+	err := r.getAllOffers(offerList)
```

Comment:
Why are you getting all the offers and not only the ones related to a specific foreign cluster?